client/lib/wp/localization/index.js

```diff
@@ -43,9 +43,17 @@ export function addLocaleQueryParam( params ) {
 		return params;
 	}
 
+	let localeQueryParam = {};
 	const query = qs.parse( params.query );
+
+	if ( params.hasOwnProperty( 'apiNamespace' ) && 'wpcom/v2' === params.apiNamespace ) {
```

Comment:
Seems like we could drop the first half of this condition and it would work just as well, since we know that it's an object and we test strictly that it equals an expected string value.

```js
if ( 'wpcom/v2' === params.apiNamespace ) {
```